src/attributes/diagnostics.md

```diff
@@ -252,10 +252,125 @@ When used on a function in a trait implementation, the attribute does nothing.
 > let _ = five();
 > ```
 
+# The `non_exhaustive` attribute
```

Comment:
Instead of doing this, I've added a new sub-section to the attributes section and added this there. I think it makes sense for this to be included alongside the other attributes, but it didn't quite fit in any of the existing categories.

Reply:
Looks good, thanks.